lifecycle-operator/controllers/options/keptnconfig_controller.go

```diff
@@ -99,6 +153,43 @@ func (r *KeptnConfigReconciler) reconcileOtelCollectorUrl(config *optionsv1alpha
 	return ctrl.Result{}, nil
 }
 
+func (r *KeptnConfigReconciler) reconcileRestApi(ctx context.Context, config *optionsv1alpha1.KeptnConfig) (ctrl.Result, error) {
+	if config.Spec.RestApiEnabled {
+		r.Log.Info("Creating Rest-Api deployment...")
+
+		err := r.Client.Create(ctx, restApiDeployment)
+		if err != nil {
+			r.Log.Error(err, "Unable to Deploy Rest API")
+			return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, err
+		}
+
+		err = r.Client.Create(ctx, restApiService)
+		if err != nil {
+			r.Log.Error(err, "Unable to Deploy Rest API Service")
+			return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, err
+		}
+
+		r.Log.Info("Created Rest-Api deployment.\n")
+		return ctrl.Result{}, nil
+	}
+
+	r.Log.Info("Deleting Rest-Api deployment...")
+	err := r.Client.DeleteAllOf(ctx, restApiDeployment)
```

Comment:
in these del errors, I'd check for `errors.IsNotFound(err)`. if it is not found, we don't need to requeue